static/js/pm_list.js

```diff
@@ -59,52 +78,157 @@ export function update_private_messages() {
 
         vdom.update(replace_content, find, new_dom, prior_dom);
         prior_dom = new_dom;
+    } else {
+        // When we update the private_messages_section in its collapsed state,
+        // the only thing to check is if there's an active conversation thread to
+        // display outside the collapsed section.
+        update_collapsed_private_messages_section();
+
+        // Also, change the toggle icon of private_messages_section to collapsed one, since
+        // we have `private_messages_collapsed` as `true`.
+        $("#toggle-private-messages-section-icon").removeClass("fa-caret-down");
+        $("#toggle-private-messages-section-icon").addClass("fa-caret-right");
     }
 }
 
 export function expand() {
-    private_messages_open = true;
+    private_messages_collapsed = false;
     stream_popover.hide_topic_popover();
     update_private_messages();
-    if (pm_list_data.is_all_privates()) {
-        $(".top_left_private_messages").addClass("active-filter");
-    }
 }
 
 export function update_dom_with_unread_counts(counts) {
     update_private_messages();
     set_count(counts.private_message_count);
 }
 
-function should_expand_pm_list(filter) {
-    const op_is = filter.operands("is");
+function update_collapsed_private_messages_section() {
+    // when we need to update the collapsed private messages section, we first check
+    // for any active conversation thread which is supposed to be present outside to it.
+    const conversations = pm_list_data.get_conversations();
+    const active_conversation = conversations.find((conversation) => conversation.is_active);
+
+    if (active_conversation) {
+        // If we find an active conversation thread satisfying the requirement, then we update our
+        // DOM with it's node.
+        const node = [pm_list_dom.keyed_pm_li(active_conversation)];
+        const new_dom = pm_list_dom.pm_ul(node);
+        const $container = ui.get_content_element($("#sidebar-private-messages-container"));
+        function replace_content(html) {
+            $container.html(html);
+        }
 
-    if (op_is.length >= 1 && op_is.includes("private")) {
-        return true;
+        function find() {
+            return $container.find("ul");
+        }
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // else mark the section as collapsed.
+        close();
     }
+}
```

Comment:
This function very deeply duplicates `update_private_messages`; I've rewritten it to not do that.